### DISCUSSION TITLE
mathtools and require in rehypeMathjax?

### DISCUSSION THREAD

**Discussion by mzietlow** [remarkjs/remark#1224] [Q&A] [answered]
Hi all,

when I try to `\require{mathtools}` (which is supported by mathjax, cf. [the docs](https://docs.mathjax.org/en/latest/input/tex/extensions/mathtools.html)), I get the following error:

``` text
Undefined control sequence \require
```

I get a similar error, when I try to use commands from the mathtools-Package, for example `\newcommand{\qed}[0]{\square}` results in

``` text
Undefined control sequence \newcommand
```

I have configured rehypeMathjax as follows:

``` ts
rehypeMathjax,
{
  loader: { load: ["[tex]/mathtools", "[tex]/color", "[tex]/require"] },
  tex: {
    inlineMath: [
      ["$", "$"],
      ["\\(", "\\)"],
    ],
    packages: { "[+]": ["mathtools", "color", "require"] },
  },
},
```

Any ideas? :)

**Comment by ChristianMurphy** [comment]
Welcome @mzietlow! 👋 
Sorry you ran into a spot of trouble.
Could you share a bit more about your setup?
I'm unclear how you are adding the extensions on the page, what environment you are running this in, and what build tools you are using.
Consider sharing a sandbox with a runnable example of what you are trying so others can have a better understanding of your setup.

Some sandbox starters:

| description | codesandbox | stackblitz |
| - | - | - |
| remark only (for markdown to markdown) | [codesandbox](https://codesandbox.io/s/github/remarkjs/.github/tree/main/sandbox-templates/remark-with-vite) | [stackblitz](https://stackblitz.com/github/remarkjs/.github/tree/main/sandbox-templates/remark-with-vite) |
| remark and rehype (for markdown to html) | [codesandbox](https://codesandbox.io/s/github/remarkjs/.github/tree/main/sandbox-templates/remark-rehype-with-vite) | [stackblitz](https://stackblitz.com/github/remarkjs/.github/tree/main/sandbox-templates/remark-rehype-with-vite) |
| react-markdown | [codesandbox](https://codesandbox.io/s/github/remarkjs/.github/tree/main/sandbox-templates/react-markdown-with-vite) | [stackblitz](https://stackblitz.com/github/remarkjs/.github/tree/main/sandbox-templates/react-markdown-with-vite) |

**Comment by mzietlow** [comment, marked as answer]
Hi @ChristianMurphy, thank you for getting me started with the sandboxes :)
I am using [quartz](https://github.com/jackyzha0/quartz) and did not have any exposure to remarkjs before.

Initially, I was using the following setup:

``` ts
const file = await unified()
  .use(remarkParse)
  .use(remarkMath)
  .use(remarkRehype)
  .use(rehypeMathjax, {
    loader: { load: ['[tex]/mathtools', '[tex]/color', '[tex]/require'] },
    tex: {
      packages: { '[+]': ['mathtools', 'color', 'require'] },
    },
  })
  .use(rehypeStringify)
  .process(sourceMarkdown);
```

But by tinkering a little, I found out that all of it was unnecessary. Unlike my experience with 

``` html
<script type="text/javascript" id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>

```

rehypeMathjax seems to include mathtools out of the box already. Thus, the following config suffices

``` ts
const file = await unified()
  .use(remarkParse)
  .use(remarkMath)
  .use(remarkRehype)
  .use(rehypeMathjax)
  .use(rehypeStringify)
  .process(sourceMarkdown);
```

And while `$\\require{mathjax}$` does _not_ work, all my markdown commands _requiring_ mathtools, e.g.,

```md
# example

$\\bigtimes$
```

do work and also `\newcommand` as in the following

```md 
$\\newcommand{\\given}[0]{\\mid}$

$P(A\\given B)$
``` 

works like a charm :)

Now I'm left wondering, why my initial configuration led to such strange behavior - but happy that it works 😌
